packages/@glimmer/runtime/lib/vm/gbox.ts

```diff
@@ -130,12 +143,12 @@ export class Context {
   private encodeObject(a: any): number {
     const idx = this.stack.length;
     this.stack.push(a);
-    return this.encodeNumberAndTag(idx, Tag.ANY);
+    return this.encodeNumberAndTag(idx, Tag.ANY, isConst(a));
```

Comment:
Oh here I think you'll want to run through the generic `encode` method again with `a.value()`, right?